sway-core/src/language/ty/expression/expression.rs

```diff
@@ -597,11 +597,10 @@ impl TyExpression {
             TyExpressionVariant::FunctionParameter => {}
             TyExpressionVariant::MatchExp {
                 desugared,
-                //scrutinees,
+                // TODO: We must also check scrutinees.
```

Comment:
Why is this left as a TODO? Isn't it just a single recursive call?

Reply:
Unfortunatelly it isn't that straightforward. I thought of doing it, but it is a bit of an effort. We have ~10 different scrutinees, each needs a different treatment, some are recursive... The whole deprication needs additional work and we have a separate issue for that, #6942. There are several TODOs in this file and I've linked them now to that issue.